capstone-backend/src/main/java/com/google/sps/servlets/Utility.java

```diff
@@ -36,26 +44,185 @@ public static Firestore getFirestoreDb() throws IOException {
+        response.sendError(HttpServletResponse.SC_NOT_FOUND);
+        return null;
       }
-    } catch (Exception e) { 
+    } catch (Exception e) {
       System.err.println("Error: " + e);
+      response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+      return null;
+    }
+    retrievedObjects.add(item);
+    return retrievedObjects;
+  }
+
+  /**
+   * Appends filters to the given query depending on fields in the request query 
+   * @param {genericClass} a Generic class, used in casting documents from the database
+   * @param {response} response returned from the call
+   * @param {it} iterator for the Map of query parameter names to values
+   * @param {query} query being modified, to eventually be used to query the database.
+   * @return Query the query being modified, to eventually be used to query the database.
+   */
+  private static <T> Query addToQuery(GenericClass<T> genericClass, HttpServletResponse response, 
+      Iterator<Map.Entry<String, String[]>> it, Query query) throws IOException {
+    Map.Entry<String, String[]> entry = it.next();
+    String parameterName = entry.getKey();
+    if (entry.getValue().length != 1) {
+      System.err.println("Error: Each parameter should have only one value");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return null;
+    }
+    if (entry.getValue()[0].isEmpty()) {
+      System.err.println("Error: The parameter value cannot be empty");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return null;
+    }
+    String parameterValue = entry.getValue()[0];
+    Field[] fields = genericClass.getMyType().getDeclaredFields();
+    boolean containsParameter = false;
+    boolean parameterIsList = false;
+    for (Field field : fields) {
+      if (field.getName().equals(parameterName)) {
+        containsParameter = true;
+        if (field.getType().getName().equals("java.util.List") || field.getType().getName().equals("java.util.ArrayList")) {
+          parameterIsList = true;
+        }
+      }
+    }
+    if (!containsParameter) {
+      System.err.println("Error: The object does not have the field specified in the request parameter.");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return null;
+    }
+    if (!parameterIsList) {
+      return query.whereEqualTo(parameterName, parameterValue);
+    }
+    else {
+      return query.whereArrayContains(parameterName, parameterValue);
+    }
+  }
+
+  /**
+   * Gets an asynchronous collection of matching documents from the database matching the request
+   * @param {collectionRefence} reference to the appropriate collection of documents in the database
+   * @param {genericClass} a Generic class, used in casting documents from the database
+   * @param {request} request sent to the backend
+   * @param {response} response returned from the call
+   * @return Query the query being modified, to eventually be used to query the database.
```

Comment:
Can you update the `@return` annotation here and reorder the `@param` annotations to match the parameters in the `getByField` method?

Reply:
Sure!